diff -up libphonenumber-8.12.7/cpp/CMakeLists.txt.1~ libphonenumber-8.12.7/cpp/CMakeLists.txt
--- libphonenumber-8.12.7/cpp/CMakeLists.txt.1~	2020-07-02 08:40:20.000000000 +0200
+++ libphonenumber-8.12.7/cpp/CMakeLists.txt	2020-07-11 02:47:54.981750109 +0200
@@ -480,7 +481,9 @@ if (${BUILD_GEOCODER} STREQUAL "ON")
   list (APPEND GEOCODER_DEPS ${COMMON_DEPS})
   # Note that the subset of base/ on which the geocoder relies is implemented
   # on top of Boost header-only libraries (e.g. scoped_ptr.hpp).
-  target_link_libraries (geocoding ${LIBRARY_DEPS})
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
+    target_link_libraries (geocoding ${LIBRARY_DEPS})
+  endif ()
   target_link_libraries (geocoding-shared ${LIBRARY_DEPS})
 endif ()
 
@@ -647,7 +650,9 @@ if (BUILD_SHARED_LIB)
 endif ()
 
 if (${BUILD_GEOCODER} STREQUAL "ON")
-  install (TARGETS geocoding LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR})
+  if (${BUILD_STATIC_LIB} STREQUAL "ON")
+    install (TARGETS geocoding LIBRARY DESTINATION ${LIBDIR} ARCHIVE DESTINATION ${LIBDIR})
+  endif ()
   install (TARGETS geocoding-shared LIBRARY DESTINATION ${LIBDIR} ARCHIVE
            DESTINATION ${LIBDIR})
 endif ()
