diff -up libphonenumber-8.13.7/cpp/CMakeLists.txt.omv~ libphonenumber-8.13.7/cpp/CMakeLists.txt
--- libphonenumber-8.13.7/cpp/CMakeLists.txt.omv~	2023-03-08 22:01:45.095428305 +0100
+++ libphonenumber-8.13.7/cpp/CMakeLists.txt	2023-03-08 22:01:51.748502702 +0100
@@ -20,7 +20,7 @@ project (libphonenumber VERSION 8.13.0)
 
 # Pick the C++ standard to compile with.
 # Abseil currently supports C++11, C++14, and C++17.
-set(CMAKE_CXX_STANDARD 11 CACHE STRING "C++ standard used to compile this project")
+set(CMAKE_CXX_STANDARD 17 CACHE STRING "C++ standard used to compile this project")
 set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_POSITION_INDEPENDENT_CODE TRUE)
 
